### PR TITLE
LPD-31852 - Navigation menu using "Use Custom Name" checkbox name is not returned by Headless Delivery API

### PR DESCRIPTION
**Pull request by ruben-pulido**
https://liferay.atlassian.net/browse/LPD-31852 - Navigation menu using "Use Custom Name" checkbox name is not returned by Headless Delivery API

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java [hunk 1]
```diff
@@ -667,54 +764,88 @@ private void _testGetNavigationMenu(
...
+		NavigationMenu getNavigationMenu =
+			navigationMenuResource.getNavigationMenu(
+				postNavigationMenu.getId());
 
-		Assert.assertEquals(nameEnUS, nameI18nMap.get("en-US"));
-		Assert.assertEquals(nameEsES, nameI18nMap.get("es-ES"));
+		_assertNavigationMenuItem(
+			nameI18nMap1.get(locale.toLanguageTag()), nameI18nMap1,
+			getNavigationMenu.getNavigationMenuItems()[0], "navigationMenu",
+			false);
```
Comment (ealonso) [inline modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java:824]: @ruben-pulido I think you need to remove empty lines

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java [hunk 2]
```diff
@@ -667,54 +764,84 @@ private void _testGetNavigationMenu(
...
+			HashMapBuilder.put(
+				LocaleUtil.US,
+				_friendlyURLNormalizer.normalizeWithEncoding(
+					StringPool.SLASH + RandomTestUtil.randomString())
+			).build());
 
-		NavigationMenu getNavigationMenu =
-			navigationMenuResource.getNavigationMenu(
-				postNavigationMenu.getId());
+		HashMap<Locale, String> layoutNameMap2 = HashMapBuilder.put(
```
Comment (ealonso) [inline modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java:784]: @ruben-pulido HashMap it should be the generic type Map

**Comment by ruben-pulido** [comment]
@ealonso Feedback incorporated and commits consolidated here: https://github.com/liferay-echo/liferay-portal/pull/16019

**Pull request closed without merging**